### PR TITLE
Start new retro

### PR DESCRIPTION
**Pull request by ampeigonet**
Story: https://trello.com/c/dNFuaK4J/2-start-a-new-retrospective

### CODE REVIEW

**Comment by fchavat** [comment]
It seems like each pull request is carrying commits from previous pull requests.
Maybe you created the branch `start-new-retro` from the top of `visit-retro-app` but you did the pull request from `start-new-retro` to `master`. If that's the case, maybe it's better to make the pull request from `start-new-retro` to `visit-retro-app` and make a rebase to `master` when `visit-retro-app` gets merged.
What do you think?

If you want to change the branch in which this pull request will be merge into, you can do it by editing it (there's no need to close this pull request and open another).

File: db/migrate/20181010172034_create_retros.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateRetros < ActiveRecord::Migration[5.2]
+  def change
+    create_table :retros do |t|
+
```
Comment (fchavat) [inline db/migrate/20181010172034_create_retros.rb]: Extra line

File: app/helpers/retros_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module RetrosHelper
```
Comment (fchavat) [inline app/helpers/retros_helper.rb]: It seems that this and other files are being auto generated by `scaffold`, but these files are not needed or implements some functionality that is not being asked in the user story (`Start a New Retrospective`).
Sometimes, when we have to implement small changes, generators like `scaffold` are too much.
What do you think?

File: app/assets/stylesheets/retros.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
+html {
+  margin: 0;
+
+  body {
+    margin: 0;
+
+    #header {
+      text-align: right;
+      margin: 0px;
+      margin: 0;
```
Comment (fchavat) [inline app/assets/stylesheets/retros.scss]: Duplicated property

File: app/assets/stylesheets/retros.scss [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          margin-top: 5px;
+        }
+      }
+    }
+
+    #buttons .button_to {
+      text-align: center;
+      margin-top: 20%;
+
+      .start-button {
```
Comment (fchavat) [inline app/assets/stylesheets/retros.scss]: It seems that these properties are already written in another file (`greetings.scss`) and because of how [the asset pipeline](https://guides.rubyonrails.org/asset_pipeline.html) works, these properties are both being applied.

File: app/assets/stylesheets/scaffolds.scss [hunk 5]
```diff
@@ -0,0 +1,84 @@
+body {
+  background-color: #fff;
+  color: #333;
+  margin: 33px;
```
Comment (fchavat) [inline app/assets/stylesheets/scaffolds.scss]: Because of the asset pipeline, all style assets are included in the views, so, if you apply the same property to the same tag but in different assets, one will overrides the other, depending on which file is firstly added by the asset pipeline.
In this case, `margin: 33px` is overriden by `margin: 0` in `retros.scss`
Maybe I am not explaining well. [This guide](https://guides.rubyonrails.org/asset_pipeline.html) will help you understand how it works.

File: app/views/greetings/hello.html.erb [hunk 6]
```diff
@@ -1,17 +1,12 @@
 <head>
+  <title>Home - Wye RetroApp</title>
```
Comment (fchavat) [inline app/views/greetings/hello.html.erb]: I think that this is not working. Maybe you can make use of [a named yield](https://guides.rubyonrails.org/layouts_and_rendering.html#understanding-yield) and the [content_for](https://guides.rubyonrails.org/layouts_and_rendering.html#using-the-content-for-method) helper to achieve this objective.
[This question](https://stackoverflow.com/questions/185965/rails-how-to-change-the-title-of-a-page) can help too!

**Review by fchavat — COMMENTED** [review]
Left some comments!

**Review by rodrigopdl — COMMENTED** [review]
The story mentions two different text buttons, "Start" and "Start New Retro!". 

However, this is an error in the story's description. The idea is that only one button exists in the context of this story, which is centered in the root page, with the text "Start". _(as mentioned by A.C.'s 1 and 2)_

Keep in mind that we're only mentioning one button in the A.C., so the "Continue" button also doesn't seem to be necessary here. 🙂

File: app/models/retro.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+class Retro < ApplicationRecord
```
Comment (rodrigopdl) [inline app/models/retro.rb]: Is the `Retro` model required to fulfill the story's A.C.?

File: app/controllers/retros_controller.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
+class RetrosController < ApplicationController
```
Comment (rodrigopdl) [inline app/controllers/retros_controller.rb]: Are all these endpoints needed to fulfill the story's A.C.?

File: app/views/retros/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,17 @@
+<%= form_with(model: retro, local: true) do |form| %>
```
Comment (rodrigopdl) [inline app/views/retros/_form.html.erb]: Are we making use of this form? Is it necessary to fulfill the story's A.C.?

File: app/views/retros/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,25 @@
+<p id="notice"><%= notice %></p>
```
Comment (rodrigopdl) [inline app/views/retros/index.html.erb]: Are these files needed at this point? `edit.html.erb``index.html.erb``_retro.json.jbuilder`

**Review by rodrigopdl — COMMENTED** [review]
Left some comments!

File: app/helpers/retros_helper.rb (hunk 2 above)
Comment (ampeigonet) [reply app/helpers/retros_helper.rb]: Great remark! This is something that we discussed yesterday with @rodrigopdl and @FLarra  regarding another story. As you mention, generating a scaffold is a little bit too much since I don't even know if we will need to persist any data in the future (or use other methods, etc), maybe a controller with a single method (new) is enough for now?

File: app/views/greetings/hello.html.erb (hunk 6 above)
Comment (ampeigonet) [reply app/views/greetings/hello.html.erb]: That's correct :D this is fixed in https://github.com/anita040593/RetroApp/pull/3 (hopefully)

But since it's not working in this PR I'll delete the line :)

File: app/helpers/retros_helper.rb (hunk 2 above)
Comment (fchavat) [reply app/helpers/retros_helper.rb]: I agree with you, I think it's enough to accomplish the a.c. of the user story!

**Comment by ampeigonet** [comment]
@rodrigopdl @fchavat  Addressed your comments! Feel free to review the code again whenever you can :)

**Review by fchavat — APPROVED** [review]
LGTM 💪!
